docs/operations/upgrade-gardener.md

```diff
@@ -0,0 +1,115 @@
+In order to upgrade them, update their OCI references in these `Extension` resources.
+
+In case you also have extensions manually deployed to the (virtual) garden cluster (not recommended), update the version in their `ControllerDeployment` resources.
+
+In both cases, make sure to always check the extension's release notes to ensure compatibility with the Gardener version.
+
+### Verify Readiness
+
+To confirm that all your seed clusters are updated, check the following for each `Seed` resource:
+
+- The `gardener.cloud/operation` annotation is removed, indicating the operation is complete.
+- The `.status.gardener.version` field shows your target version.
+- The `.status.observedGeneration` matches the `.metadata.generation`, meaning the latest configuration has been processed.
+- The `.status.lastOperation.state` is `Succeeded`.
+
+Finally, check the health conditions (`.status.conditions[]`) in the `Seed` resources to ensure they all report `True`.
+
+## 4. Shoot Reconciliations
+
+By default, after a `gardenlet` is upgraded, it immediately starts reconciling the `Shoot` clusters it manages. While this is suitable for small setups, it's often better to perform these reconciliations only during a predefined maintenance window.
+
+To enable this, set `.controllers.shoot.reconcileInMaintenanceOnly=true` in the `gardenlet`'s [component configuration](../../example/20-componentconfig-gardenlet.yaml). When this setting is enabled, all `Shoot`s will be reconciled during their next scheduled maintenance window, which typically occurs within 24 hours. You can learn more about shoot maintenance [here](../usage/shoot/shoot_maintenance.md#cluster-reconciliation).
+
+> [!IMPORTANT]
+> Make sure that all `Shoot`s have been reconciled (even if it failed) with the currently installed Gardener version before upgrading to the next version.
+> This is especially relevant if you choose to only reconcile in the shoot clusters' maintenance time windows.
+> In this case, you need to wait `24h` before upgrading Gardener again to be on the safe side.
+>
+> You can verify that all `Shoot`s have been reconciled with the current Gardener version by chekcing that their `.status.gardener.version` fields show your target version.
+
+### Operating System Config Updates
+
+Similar to seed upgrades, updates to the operating system on `Shoot` cluster worker nodes are also staggered. This prevents all nodes from being updated simultaneously, which could cause disruptions like `kubelet` restarts across the entire cluster.
+
+By default, the rollout across all nodes completes within 5 minutes. You can customize this timeframe by adding the `shoot.gardener.cloud/cloud-config-execution-max-delay-seconds` annotation to your `Shoot`s. A value of `0` updates all nodes in parallel, while a higher value spreads the update over a longer period (up to 1800 seconds).
+
+## Image Vector and Overwrites
```

Comment:
Should we move/reference this to/in the second step? Updating the overwrites needs to be done as a preparation